tests/test_routes.py

```diff
@@ -351,6 +351,104 @@ def test_create_valid_customer_valid_address_pincode(self):
             created_addr = addr_post_req.get_json()
             self.assertEqual(created_addr["pin_code"], addr.pin_code, "Addresss pincodes has not been populated correctly")
 
+    def test_get_customer(self):
+        """It should Read a single Customer"""
+        # get the id of an customer
+
+        customer = CustomerFactory()
+        #create the customer
+        resp= self.client.post(BASE_URL, json=customer.serialize())
+        self.assertEqual(
+            resp.status_code, 
+            status.HTTP_201_CREATED,
+            "Could not create test Customer"
+        )
+        new_customer = resp.get_json()
+        customer.id = new_customer["id"]
+        resp2 = self.client.get(
+            f"{BASE_URL}/{customer.id}", content_type="application/json"
+        )
+        self.assertEqual(resp2.status_code, status.HTTP_200_OK)
+        data = resp2.get_json()
+        self.assertEqual(data["id"], customer.id)
+        self.assertEqual(data["first_name"], customer.first_name)
```

Comment:
Are we not checking all the attributes of the customer? Last name, email etc? 

Might want to maintain parity with address since you are checking all the attributes while testing for address?

Reply:
Yes, good catch. Fixed this.